tests/models_test.py

```diff
@@ -0,0 +1,149 @@
+                    assert unobserved['is_exposed'] in (0, 1)
+                    # For how long has been exposed
+                    assert 0 <= unobserved['exposure_day'] < 14
+                    # Is infectious or not
+                    assert unobserved['is_infectious'] in (0, 1)
+                    # For how long has been infectious
+                    assert 0 <= unobserved['infectious_day'] < 14
+                    # Is recovered or not
+                    assert unobserved['is_recovered'] in (0, 1)
+                    # For how long has been infectious
+                    assert 0 <= unobserved['recovery_day'] < 14
+                    # Locations where unobserved['is_exposed'] was true
+                    assert len(unobserved['exposed_locs'].shape) == 1
+                    assert unobserved['exposed_locs'].min() in (0, 1)
+                    assert unobserved['exposed_locs'].max() in (0, 1)
+                    assert 0 <= unobserved['exposed_locs'].sum() <= len(unobserved['exposed_locs'])
+                    # Encounters responsible for exposition. Exposition can occur without being
+                    # linked to an encounter
+                    assert len(unobserved['exposure_encounter'].shape) == 1
+                    assert unobserved['exposure_encounter'].min() in (0, 1)
+                    assert unobserved['exposure_encounter'].max() in (0, 1)
+                    assert unobserved['exposure_encounter'].sum() in (0, 1)
+                    # Level of infectiousness / day
+                    assert unobserved['infectiousness'].shape == (14,)
+                    assert unobserved['infectiousness'].min() >= 0
+                    assert unobserved['infectiousness'].max() <= 1
+
+                    # observed['reported_symptoms'] is a subset of unobserved['true_symptoms']
+                    assert (unobserved['true_symptoms'] == observed['reported_symptoms']) \
+                           [observed['reported_symptoms'].astype(np.bool)].all()
+
+                    if unobserved['is_infectious'] or unobserved['is_recovered']:
+                        assert unobserved['is_infectious'] != unobserved['is_recovered']
+
+                    # exposed_locs is the same length as candidate_locs
+                    # TODO: observed['candidate_locs'] should be a tuple (human_readable, id) preferably sorted
+                    assert unobserved['exposed_locs'].shape == (len(observed['candidate_locs']),)
+
+                    # exposure_encounter is the same length as candidate_encounters
+                    assert unobserved['exposure_encounter'].shape == (observed['candidate_encounters'].shape[0],)
+
+                    if prev_observed:
+                        assert (observed['reported_symptoms'][:13, :] == prev_observed['reported_symptoms'][-13:, :]).all()
+                        assert (observed['candidate_encounters'][observed['candidate_encounters'][:, 2] > 1][:, 0:2] ==
```

Comment:
this is currently `False` but I'm not sure why. It seams that `observed['candidate_encounters'][:, 2]` does not update with + 1 compared with `prev_observed['candidate_encounters'][:, 2]`